installer/pkg/config/v1/config.go

```diff
@@ -242,14 +245,14 @@ type BlockNewUsers struct {
 }
 
 type OAuth struct {
-	ClientId            string            `json:"clientId"`
-	ClientSecret        string            `json:"clientSecret"`
-	CallBackUrl         string            `json:"callBackUrl"`
+	ClientId            string            `json:"clientId" validate:"required"`
+	ClientSecret        string            `json:"clientSecret" validate:"required"`
+	CallBackUrl         string            `json:"callBackUrl" validate:"required"`
 	AuthorizationUrl    string            `json:"authorizationUrl"`
 	TokenUrl            string            `json:"tokenUrl"`
 	Scope               string            `json:"scope"`
 	ScopeSeparator      string            `json:"scopeSeparator"`
-	SettingsUrl         string            `json:"settingsUrl"`
+	SettingsUrl         string            `json:"settingsUrl" validate:"required"`
```

Comment:
In looking at the config map for core-dev, the `SettingsUrl` does not exist for BitBucket. 

Is it worth using [a custom validator](https://github.com/go-playground/validator/issues/494), to only make this required if the type is GitHub or GitLab?

Reply:
Good catch. It's not actually a required property according to the [TypeScript interface](https://github.com/gitpod-io/gitpod/blob/b54521de75b143db4a289918b2a9d4a9e96ffca8/components/gitpod-protocol/src/protocol.ts#L1152)

Reply:
Gotcha, thank you very much for also sharing the corresponding typescript interface! :100: